aiobotocore/httpsession.py

```diff
@@ -99,14 +102,30 @@ def __init__(
         # request so don't need proxy manager
 
     async def __aenter__(self):
-        assert not self._sessions
+        with _session_lock:
+            self._aexited = False
 
+        assert not self._sessions
         return self
 
     async def __aexit__(self, exc_type, exc_val, exc_tb):
         self._sessions.clear()
```

Comment:
there's an easier fix, just set self._sessions to None, then you don't need a lock